File: API/Entities/Memo.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace API.Entities;

public class Memo
{
    public int Id { get; set; }

    public string? Title { get; set; }

    [Required]
    public string Content { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public bool IsPublic { get; set; } = false;
    public string? MediaUrls { get; set; }

    // Foreign keys
    public required string AppUserId { get; set; }

    // Navigation properties
    public AppUser AppUser { get; set; } = null!;
}

File: client/src/types/memo.ts
export interface Memo {
    id: number;
    content: string;
    createdAt: string;
    updatedAt: string;
    isPublic: boolean;
    mediaUrls?: string[];
}

export interface CreateMemo {
    content: string;
    isPublic?: boolean;
    mediaUrls?: string[];
}

export interface UpdateMemo {
    content?: string;
    isPublic?: boolean;
    mediaUrls?: string[];
}